var/spack/repos/builtin/packages/cudnn/package.py

```diff
@@ -32,12 +32,26 @@ class Cudnn(Package):
 
     homepage = "https://developer.nvidia.com/cudnn"
 
+    version('7.0', '6189e05077dbce0bd07059b20306a836',
+            url='http://developer.download.nvidia.com/compute/redist/cudnn/v7.0.3/cudnn-9.0-linux-x64-v7.tgz',
+            when='^cuda@9')
```

Comment:
@weijianwen I don't know if there's any effect if you add a `when` argument to a `version` - what was the effect you wanted?